src/lib/task.js

```diff
@@ -696,11 +696,20 @@ export class Task extends EventEmitter {
       // we don't wait for the promise to resolve just trigger kill here which
       // will cause run to stop processing the task and give us an error
       // exit code.
-      this.dockerProcess.kill();
-      this.stream.write(fmtErrorLog(
-        'Task timeout after %d seconds. Force killing container.',
-        this.task.payload.maxRunTime
-      ));
+      // But first we kill with SIGINT, giving the process a chance to cleanup
+      setTimeout(() => {
+        this.dockerProcess.kill();
+        this.stream.write(fmtErrorLog(
+          'Task timeout after %d seconds. Force killing container.',
+          this.task.payload.maxRunTime
+        ));
+      }, 5 * 1000);
+      try {
+        this.dockerProcess.container.kill({signal: 'SIGINT'}).catch(err =>
```

Comment:
I think propagation and graceful termination is the responsibility of the task-author...
Notably: we should **not** be sending `SIGINT` to all subprocesses, because it is the job of the task-author to gracefully stop...

If the task script doesn't gracefully stop, which could involve stopping some processes before others, etc... then we will force kill..

If we send SIGINT to every subprocess the root process won't be able to make a controlled graceful termination, because we might have killed resources it depended on for that purposes.